xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -484,7 +484,7 @@ private static LdsUpdate processServerSideListener(Listener listener)
     if (proto.getQueryParametersCount() != 0) {
       return null;
     }
-    StructOrError<PathMatcher> pathMatch = parsePathMatcher(proto);
+    StructOrError<RouteMatcher> pathMatch = parseRouteMatcher(proto);
```

Comment:
This change is misleading though. The matcher only matches the request path, while matching with a Route includes a lot of other components to be matched.

Update: Oh, I see you are mentioning merging RouteMatcher with PathMatcher.